extensions-core/s3-extensions/src/main/java/io/druid/firehose/s3/StaticS3FirehoseFactory.java

```diff
@@ -255,4 +258,33 @@ public int hashCode()
   {
     return S3Utils.S3RETRY;
   }
+
+  @Override
+  public FiniteFirehoseFactory<StringInputRowParser, URI> withSplit(InputSplit<URI> split)
+  {
+    return new StaticS3FirehoseFactory(
+        s3Client,
+        Collections.singletonList(split.get()),
+        null,
+        getMaxCacheCapacityBytes(),
+        getMaxFetchCapacityBytes(),
+        getPrefetchTriggerBytes(),
+        getFetchTimeout(),
+        getMaxFetchRetry()
+    );
+  }
+
+  private static URI toUri(S3ObjectSummary object)
```

Comment:
This is something it would be nice if we didn't have to maintain. Are there any methods in the Amazon java SDK that will construct a URI or URL for us?

Reply:
If not please document the method as such and why we have to roll our own.

Reply:
I already tried, but couldn't find any. Added a document.